internal/fs/fs.go

```diff
@@ -914,13 +913,21 @@ func (fs *fileSystem) lookUpLocalFileInode(parent inode.DirInode, childName stri
 		child = fs.localFileInodes[fileName]
 
 		if child == nil {
+			fs.mu.Unlock()
 			return
 		}
 
 		// If the inode already exists, we need to follow the lock ordering rules
 		// to get the lock. First get inode lock and then fs lock.
 		fs.mu.Unlock()
 		child.Lock()
```

Comment:
I would suggest taking fs.mu.lock here instead of carefuly unlocking in every exit which can get skipped in future edits.

Just add a comment that we are taking the lock early to be able to use common defer function.

Reply:
Done